sql-plugin/src/main/java/com/nvidia/spark/rapids/UnsafeRowToColumnarBatchIterator.java

```diff
@@ -135,13 +143,9 @@ public ColumnarBatch next() {
         // Grab the semaphore because we are about to put data onto the GPU.
         TaskContext tc = TaskContext.get();
         if (tc != null) {
-          GpuSemaphore$.MODULE$.acquireIfNecessary(tc);
-        }
-        if (totalTime != null) {
-          buildRange = new NvtxWithMetrics("RowToColumnar", NvtxColor.GREEN, totalTime);
-        } else {
-          buildRange = new NvtxRange("RowToColumnar", NvtxColor.GREEN);
+          GpuSemaphore$.MODULE$.acquireIfNecessary(tc, semaphoreWaitTime);
         }
+        buildRange = NvtxWithMetrics.apply("RowToColumnar: build", NvtxColor.GREEN, Option.apply(gpuOpTime));
```

Comment:
```suggestion
        buildRange = NvtxWithMetrics.apply("RowToColumnar: build", NvtxColor.GREEN, Option(gpuOpTime));
```

Reply:
Because this is Java code, we have to explicity call the `apply` method here.